lms/djangoapps/instructor_task/tasks_helper/utils.py

```diff
@@ -45,3 +53,48 @@ def tracker_emit(report_name):
     Emits a 'report.requested' event for the given report.
     """
     tracker.emit(REPORT_REQUESTED_EVENT_NAME, {"report_type": report_name, })
+
+
+class SensitiveMessageOnReports(object):
+    """
+    Adds a sensitive data message to the reports if the flag
+    DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS is enabled.
+
+    Due to reports being generated in different ways, each one handles their own way:
+        1. Using the CSV library directly.
+        2. Using upload_csv_to_report_store function which in turn
+           uses store_rows of DjangoStorageReportStore class where is built the CSV.
+    """
+
+    def __init__(self):
+        self.should_msg_be_displayed = configuration_helpers.get_value(
+            'DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS',
+            settings.FEATURES.get(
+                'DISPLAY_SENSITIVE_DATA_MSG_FOR_DOWNLOADS',
+                False
+            )
+        )
+
+    def csv_direct(self, writer):
+        """
+        Writes the row immediately in the CSV.
+        """
+        if self.should_msg_be_displayed:
+            msg = self.process_message()
+            encode = unicode(msg).encode('utf-8')
+            return writer.writerow([encode])
+
+    def with_report_store(self):
+        """
+        Return the string parsed in process_message. This string is passed
+        to upload_csv_to_report_store which decides if execute it or not.
+        """
+        return [self.process_message()]
+
+    def process_message(self):
```

Comment:
@jagonzalr Sorry for not catching that earlier but this will only produce message in one language, no?
As far as I understand the tech doc, we need it in three languages here. It might be worth adding another setting variable with the list of languages in which we need to display the disclaimer and then update `process_message` to produce the message in all requested languages.

Reply:
@jagonzalr Yes, you're right that it will translate the message :) It will though be then included in the CSV file in one language (according to the language of the website). What I was suggesting is that - according to the Tech doc - we need the disclaimer to be included in 3 languages at the same time, no matter what language of the website user chooses.

Reply:
I'm reading again the document and I can't see this @tomaszgy. But I think we change it to not include the 3 languages because it would be very hard to contribute this back to edX.

I can't remember if that was the case @pomegranited?

Reply:
If that is the case, then I'm sorry, I must have missed the changes in the doc. Will ask Anna on the ticket to make sure.

Reply:
Don't worry @tomaszgy maybe I'm the one mistaken jaja